util/design/gen-flash-img.py

```diff
@@ -12,86 +12,167 @@
+    mask = FLASH_GF_2_64.Multiply(operand_a, operand_b)
     masked_data = data ^ mask
     return prince.prince(masked_data, flash_data_key,
-                         PRINCE_NUM_HALF_ROUNDS) ^ mask
+                         FLASH_PRINCE_NUM_HALF_ROUNDS) ^ mask
 
 
-def main(argv: List[str]):
-    # Parse command line args.
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--infile",
-                        "-i",
-                        type=str,
-                        help="Input VMEM file to reformat.")
-    parser.add_argument("--outfile", "-o", type=str, help="Output VMEM file.")
-    parser.add_argument("--scramble",
-                        "-s",
-                        action="store_true",
-                        help="Whether to scramble data or not.")
-    parser.add_argument("--address-key",
-                        type=str,
-                        help="Flash address scrambling key.")
-    parser.add_argument("--data-key",
-                        type=str,
-                        help="Flash address scrambling key.")
-    args = parser.parse_args(argv)
+def _convert_array_2_int(data_array: List[int],
+                         data_size: int,
+                         little_endian=True) -> int:
+    """Converts array of data blocks to an int."""
+    reformatted_data = 0
+    if not little_endian:
+        data_array.reverse()
+    for i, data in enumerate(data_array):
+        reformatted_data |= (data << (i * data_size))
+    return reformatted_data
 
-    # Validate command line args.
-    if args.scramble:
-        if args.address_key is None or args.data_key is None:
-            logging.error(
-                "Address and data keys must be provided when scrambling is"
-                "requested.")
 
-    # Open original VMEM for processing.
+def _get_flash_scrambling_configs(otp_vmem_file: str,
+                                  configs: FlashScramblingConfigs) -> None:
+    # Open OTP VMEM file and read into memory, skipping comment lines.
     try:
-        orig_vmem = Path(args.infile).read_text()
+        otp_vmem = Path(otp_vmem_file).read_text()
     except IOError:
-        raise Exception(f"Unable to open {args.infile}")
-
-    # Search for lines that contain data, skipping other comment lines.
-    orig_vmem_lines = re.findall(r"^@.*$", orig_vmem, flags=re.MULTILINE)
+        raise Exception(f"Unable to open {otp_vmem}")
+    otp_vmem_lines = re.findall(r"^@.*$", otp_vmem, flags=re.MULTILINE)
+
+    # TODO: Retrieve partition with the flash scrambling enablement flags.
+    configs.scrambling_enabled = False
+
+    # Retrieve SECRET1 partition which contains the flash scrambling key seeds,
+    # stripping ECC bits from each data word when processing.
+    data_blocks_64bit = []
+    data_block_64bit = 0
+    idx = 0
+    for line in otp_vmem_lines:
+        if OTP_SECRET1_RE.search(line):
+            otp_data_word_w_ecc = int(line.split()[1], OTP_WORD_SIZE)
```

Comment:
The second argument to `int()` is not a size, but a base which defaults to 10.

As I recall, vmem files are encoded in hex, so you probably want a literal `16` here.

Reply:
oops good catch.